File: bshop/src/main/java/com/example/bshop/model/Utilisateur.java
package com.example.bshop.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "utilisateur")
public class Utilisateur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false)
    private String motDePasse;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private String prenom;

    @Column(length = 20)
    private String telephone;

    @Column(length = 255)
    private String adresse;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Role role;

    @Column(nullable = false)
    private String provider;

    @Column(name = "provider_id")
    private String providerId;

    @Column(name = "date_creation")
    private LocalDateTime dateCreation;
}

File: fshop/src/model/utilisateur.ts
export interface Utilisateur{
    id?:number,
    nom :string,
    prenom:string;
    motDePasse:string,
    email :string,
    telephone:string;
    provider:string,
    providerId:string,
    adresse:string,
   dateCreation ?:Date,
    role :Role
}

export enum Role{
    ADMIN = 'ADMIN',
    SUPER_ADMIN ='SUPER_ADMIN',
    CLIENT='CLIENT'
}